### DISCUSSION TITLE
amazing game! and suggestions!

### DISCUSSION THREAD

**Discussion by Oelda** [CE-Studio/Clonecraft#17] [General]
I just checked this game out, and I'm honestly blew away by it. I didn't expect such high quality when I started it, so overall, great job with this, whoever is working on this is doing an AMAZING job. 

side note: I would love to see a dimension in this game! something like the aether mod from minecraft I feel would really fit the theme of the game, at least in my opinion. 

another side note: something along the lines of a quest system that is also like a tutorial would be great as well! something like the stoneblock mod from minecraft could work! again, that's just in my opinion.

for reference, you could check out [Quest System - Godot Devlog #6](https://www.youtube.com/watch?v=a_tLYW5Igr0) from the channel [Eder Game Dev](https://www.youtube.com/@edergamedev), of course it's sadly just a showcase, not a tutorial but it could help!

anyways, have a nice day, and keep up the great work, this game is great so far and I can't wait to see what else is added. thanks for reading!

**Comment by Oelda** [comment]
I took a screenshot, then fully tabbed out, when I did so, it wouldn't let me load back into the game sadly
![image](https://github.com/CE-Studio/Clonecraft/assets/156653478/e02defb1-1f61-48fa-9db7-e320e1cf63c4)

**Comment by clarence112** [comment]
There's some weird bugs with the pause system that I need to figure out... Wich is weird because it *used* to work fine, and I haven't touched it.

Also I see that staircase there that I'm guessing you used to get onto the roof. You *do* have creative mode flight that you can use ;)

**Comment by Oelda** [comment]
I HAVE FLIGHT? I WISH I KNEW, I CAN'T TELL YOU THE AMOUNT OF TIMES I FELL LOL, WHAT IS THE KEYBIND IF THAT'S OKAY TO ASK?  

side note, I use the word fully tab out as in, `Show Desktop button` on windows, if that's the correct term. if I tab out, it works fine, if I fully tab out, then it won't let me tab back in. 

another bug, if you place a block in your players head, you become soft locked, you can't move, but can place/break blocks besides the one YOU'RE STUCK IN.

**Comment by clarence112** [comment]
Double tap spacebar

Haven't actually tried pressing that button, I'll look into it when I get out of work

Already known, just trying to figure out the best way to go about fixing it

**Comment by Oelda** [comment]
Thank you!

Please let me know if you can replicate the bug, as long as you don't mind that is. 

I hope the best with fixing the bug! 

Side note, when will the next update be? and what all will be added? if that's okay to ask.

**Comment by clarence112** [comment]
Next update? It hasn't even had the *first* update yet lol. The current "update" strategy is just pushing whatever changes I made whenever I made them. 

My current project is documenting all the code so that people who want to do things with it can actually understand what's going on. After that saving/loading worlds is next, then a proper inventory, etc... You'll be able to download and play it immediately as soon as it's developed.

**Reply by Oelda** [reply]
I'm sorry for the late response, I was building this lol.
![image](https://github.com/CE-Studio/Clonecraft/assets/156653478/fbcbfbcf-9aef-4382-9a81-a6b17966fff1)

But I hope nothing but the best with that, it must be a hassle to document all the code, but should be rewarding for devs, and even more so newer devs. also, how long do you think it'll take for the save/load system to be out? And how often would you like to push updates out? if that's okay to ask

**Reply by clarence112** [reply]
I have absolutely zero schedule so I can't really make any promises on timeframes, but... a decent chunk of saving and loading is already done for me by the voxel engine and I just gotta give it a ui.

**Reply by Oelda** [reply]
Do you plan on having a schedule?

**Reply by clarence112** [reply]
Doubt I'd be able to stick to it. Mostly making this project for fun and a f I tried to stick to a schedule it'd probably take most of the fun out of it, unfortunately.

**Reply by Oelda** [reply]
I see, that makes sense. Hopefully there'll be more updates soon, as this game has a lot of potential, plus I'd love to see where it goes! [ as long as it doesn't cause burnout or the fun to be taken out of it. ]  But what are you most excited to implement?

**Reply by clarence112** [reply]
Honestly all my ideas for first-party mods

Something like scrap mechanic, something like computercraft, something like create, a story mode...

There will def be enough content for me to keep updating this for years

**Reply by Oelda** [reply]
I'm sorry for the late response, and what would story mode even look like? that's an interesting topic I feel like I haven't seen done. 

but what do you mean by first-party mods? if that's okay to ask

**Reply by clarence112** [reply]
"first-party mods" translates to "mods made by the game's developer"

I intend to have most of my ideas as mods that can be easily enabled and disabled, so players can tweak the gameplay however they see fit.

Story mode would probably be some kind of quest system. I don't have enough details squared away for me to be able to say much, and anything I *could* say would be pretty heavy spoilers...

**Reply by Oelda** [reply]
I see, thank you for explaining. and what do you feel needs to be added before the game is done? and start working on the mod aspect of it? or do you plan to work on both at the same time?

and I think the idea of a quest system would be great honestly, I look forward to story mode/quest system as it'll be really interesting and engaging. [ IMO, and of course, as long as it doesn't get scrapped that is ]

**Reply by clarence112** [reply]
I'd say the game needs a functional survival mode before I'd call it "playable". Ultimately though, even the base gameplay is treated like a mod by the game, so other mod stuff is gonna come into play alongside it.

Story/quest mode is absolutely happening and will never be scrapped. There's even already a little hint hidden in the titlescreen's background, even though none of its mechanics are implemented yet.

**Reply by Oelda** [reply]
I see, that's interesting, what features survival would have? Would there be things like spears? or a forge?
But do you plan on slowly implementing the Storymode along side mods/main game? or will Storymode be like a mod/random surprise update?


`random questions lol`
What would you like the first mod you make to be? 

Are you working on any mods for it as of this moment?

do you think we'll get weather? or will we just be getting biomes? I think the idea of snow coming down and slowly turning leaves/grass/wood/etc into more snowy versions would be interesting, and beautiful.

**Reply by clarence112** [reply]
Survival is basically just going to be the absolute bare minimum. Basic blocks, tools, animals, enemies, and crafting, but not much else. Everything else, including story mode, is going to be split across mods. I don't really have any plan on what order I'm going to make them in or what's going to be first.

Seasons, while they'd be nice to have, are a bit unlikely at the current moment. I think I know how I'd go about it, but other things take priority.

**Comment by clarence112** [comment]
(gonna convert this issue to a discussion since I didn't realize those weren't enabled)
